### PR TITLE
Dashboard pending requests

### PR DESCRIPTION
**Pull request by mullinsand**
This PR started small but became much bigger as I tried to actually connect the FE to the BE. I've made several adjustments to FE services/facades and updated the index page logic and friends controller logic. All tests are currently passing with the version of the BE I have which is the most up to date I believe. The biggest thing is we need to do more sad path testing/edge case testing on our searches because we you initially start out at the app you will have zero friends and the code currently can't handle that. Sorry for another big PR, I just started finding more and more things that needed fixing.

### CODE REVIEW

File: app/facades/user_facade.rb [hunk 1]
```diff
@@ -5,14 +5,17 @@ def self.find_user(google_id)
   end
 
   def self.search(email)
-    UserService.search(email)[:data].map do |data|
-      UserPoro.new(data)
-    end
+    user_data = UserService.search(email)[:data]
+    UserPoro.new(user_data) unless user_data.empty?
```
Comment (mballantyne3) [inline app/facades/user_facade.rb]: I like this addition to consider a new/non-existent user.

File: app/services/user_service.rb [hunk 2]
```diff
@@ -6,12 +6,22 @@ def self.find_user(google_id)
...
+    JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def self.create(user)
+    response = DatabaseService.conn.post 'api/v1/users' do |req|
+      req.headers[:google_id] = user.google_id
+      req.headers[:name] = user.name
+      # .gsub(' ','_')
+      req.headers[:email] = user.email
+    end
```
Comment (mballantyne3) [inline app/services/user_service.rb:31]: this might be doing the same thing as the `find_or_create_by_auth(auth_data) method I have in the User model. This actually might be better to keep so we don't risk duplicating users on the frontend and backend
